### PR TITLE
feat: add no depeg redemption policy

### PR DESCRIPTION
**Pull request by manimal1**

### CODE REVIEW

File: subgraphs/enzyme-core/schema.graphql [hunk 1]
```diff
@@ -998,6 +1010,15 @@ type AddressList @entity {
   items: [Bytes!]!
 }
 
+type AssetConfig @entity {
+  id: ID!
+  address: Bytes!
+  referenceAddress: Bytes!
+  deviationToleranceInBps: Bytes!
```
Comment (iherger) [inline subgraphs/enzyme-core/schema.graphql:1017]: This should be `Int`

File: subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ensureNoDepegOnRedeemSharesForSpecificAssetsPolicy } from '../../entities/NoDepegOnRedeemSharesForSpecificAssetsPolicy';
+import { FundSettingsUpdated } from '../../generated/contracts/NoDepegOnRedeemSharesForSpecificAssetsPolicy4Events';
+
+export function handleFundSettingsSet(event: FundSettingsUpdated): void {
+  let policy = ensureNoDepegOnRedeemSharesForSpecificAssetsPolicy(event.params.comptrollerProxy, event.address, event);
+  policy.assetConfigs = event.params.assetConfigs;
```
Comment (iherger) [inline subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts:6]: This won't work. You will need to decode `event.params.assetConfigs` and then create the `AssetConfig` entities.

File: subgraphs/enzyme-core/contexts/ethereum.ts [hunk 3]
```diff
@@ -129,6 +129,7 @@ export const ethereum: Context<Variables> = {
         minAssetBalancesPostRedemptionPolicyAddress: suluDeployment.contracts.MinAssetBalancesPostRedemptionPolicy,
         minMaxInvestmentPolicyAddress: suluDeployment.contracts.MinMaxInvestmentPolicy,
         minSharesSupplyFeeAddress: suluDeployment.contracts.MinSharesSupplyFee,
+        noDepegOnRedeemSharesForSpecificAssetsPolicyAddress: suluDeployment.contracts.NoDepegOnRedeemSharesForSpecificAssetsPolicy,
```
Comment (iherger) [inline subgraphs/enzyme-core/contexts/ethereum.ts:132]: Use the actual address, we are no longer using the import from the environment.

File: subgraphs/enzyme-core/contexts/polygon.ts [hunk 4]
```diff
@@ -128,6 +128,7 @@ export const polygon: Context<Variables> = {
         minAssetBalancesPostRedemptionPolicyAddress: polygonV4Deployment.contracts.MinAssetBalancesPostRedemptionPolicy,
         minMaxInvestmentPolicyAddress: polygonV4Deployment.contracts.MinMaxInvestmentPolicy,
         minSharesSupplyFeeAddress: polygonV4Deployment.contracts.MinSharesSupplyFee,
+        noDepegOnRedeemSharesForSpecificAssetsPolicyAddress: polygonV4Deployment.contracts.NoDepegOnRedeemSharesForSpecificAssetsPolicy,
```
Comment (iherger) [inline subgraphs/enzyme-core/contexts/polygon.ts:131]: Same here.

File: subgraphs/enzyme-core/schema.graphql
Comment (iherger) [inline subgraphs/enzyme-core/schema.graphql:1]: Also update the enum `PolicyType`

File: subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { ensureNoDepegOnRedeemSharesForSpecificAssetsPolicy } from '../../entities/NoDepegOnRedeemSharesForSpecificAssetsPolicy';
+import { FundSettingsUpdated } from '../../generated/contracts/NoDepegOnRedeemSharesForSpecificAssetsPolicy4Events';
+
+export function handleFundSettingsSet(event: FundSettingsUpdated): void {
```
Comment (iherger) [inline subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts:4]: The function name needs to be consistent with the event type, i.e. `handleFundSettingsUpdated`

**Review by iherger — CHANGES_REQUESTED** [review]

File: subgraphs/enzyme-core/schema.graphql (hunk 1 above)
Comment (manimal1) [reply subgraphs/enzyme-core/schema.graphql:1017]: done

File: subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts (hunk 5 above)
Comment (manimal1) [reply subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts:4]: done

File: subgraphs/enzyme-core/schema.graphql
Comment (manimal1) [reply subgraphs/enzyme-core/schema.graphql:1]: done

File: subgraphs/enzyme-core/contexts/ethereum.ts (hunk 3 above)
Comment (manimal1) [reply subgraphs/enzyme-core/contexts/ethereum.ts:132]: done

File: subgraphs/enzyme-core/contexts/polygon.ts (hunk 4 above)
Comment (manimal1) [reply subgraphs/enzyme-core/contexts/polygon.ts:131]: done

File: subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts (hunk 2 above)
Comment (manimal1) [reply subgraphs/enzyme-core/mappings/v4/NoDepegOnRedeemSharesForSpecificAssetsPolicy.ts]: done

**Review by KedziaPawel — APPROVED** [review]

**Review by iherger — APPROVED** [review]

**Review by iherger — APPROVED** [review]

**Pull request merged**